test/preview/dataclasses/test_document.py

```diff
@@ -282,19 +282,18 @@ def test_from_dict_with_flat_meta():
 
 @pytest.mark.unit
 def test_from_dict_with_flat_and_non_flat_meta():
-    with pytest.raises(TypeError):
```

Comment:
This test was passing but I think for the wrong reason. Let me know if we want this case to fail, I will add an explicit type error before calling `super()`

Reply:
Maybe is best if it fails? I would keep the dictionary in line with the `__init__` signature.

Reply:
Updated the code